src/tasks/Publish/upload-package.js

```diff
@@ -9,18 +9,55 @@ export default async function uploadPackage(pkg, pkgPath, registry) {
   const gitpkgPackageName = getGitTagName(pkg);
   await execLikeShell("git init", pkgTempDirPkg);
   await execLikeShell("git add .", pkgTempDirPkg);
-  await execLikeShell("git commit -m gitpkg", pkgTempDirPkg);
+  await execLikeShell("git commit -m a", pkgTempDirPkg);
   await execLikeShell(`git remote add origin ${registry}`, pkgTempDirPkg);
-  await execLikeShell(`git tag ${gitpkgPackageName}`, pkgTempDirPkg);
 
-  // change Relive; first delete the tag
-  try {
-    await execLikeShell(
-      `git push --delete origin ${gitpkgPackageName}`,
-      pkgTempDirPkg
-    );
-  } catch (e) {
-    console.warn(e);
+  // <Relive>
+  await execLikeShell(`git fetch --all`, pkgTempDirPkg);
+  await execLikeShell(`git fetch --tags`, pkgTempDirPkg);
+
+  const { stdout } = await execLikeShell(
+    `git ls-remote --tags origin | grep refs/tags/${gitpkgPackageName}`
+  );
+
+  if (stdout) {
+    const changed = await new Promise(async (resolve, reject) => {
+      try {
+        // Exits with code 1 when changes are available
+        const changes = await execLikeShell(
+          `git diff ${gitpkgPackageName} --quiet`,
+          pkgTempDirPkg
+        );
+
+        if (changes.code === 0) {
+          resolve(false);
+        }
+      } catch (e) {
+        if (e.code === 1) {
+          resolve(true);
+          return;
+        }
+        reject(err);
+      }
+    });
+
+    if (!changed) {
+      console.log("No changes detected");
+      return;
+    }
+
+    try {
+      await execLikeShell(
+        `git push --delete origin ${gitpkgPackageName}`,
+        pkgTempDirPkg
+      );
+      await execLikeShell(`git tag -d ${gitpkgPackageName}`, pkgTempDirPkg);
```

Comment:
Why is this here? And why is `await execLikeShell(`git tag ${gitpkgPackageName}`, pkgTempDirPkg);` moved down?

Shouldn't the only change necessary be the `git diff ${gitpkgPackageName} --quiet` you added?

Reply:
So in order to diff the changes we need the tag inside our local git. And for that I needed to change the order of commands a bit:
1. Fetch all tags from remote
2. Check if the tag already exists (if not, skip step 3&4)
3. If the tag exists we can diff the new local changes with the tag
4. Because we fetched all tags from the remote we need to delete the remote and local tag otherwise we cannot publish on the same tag.
5. Push

Reply:
got it, maybe copy your step 1-4 explanation to code. but nice job!